src/genie/libs/parser/asa/show_version.py

```diff
@@ -0,0 +1,856 @@
+        p33 = re.compile(r'^Cluster\s*: (?P<cluster>\S+)')
+
+        # Serial Number: 9A5BHB00D2D
+        p34 = re.compile(r'^Serial.+: (?P<serial_number>\S+)')
+
+        # Image type          : Release
+        p35 = re.compile(r'^Image type\s*: (?P<image_type>\S+)')
+
+        # Key version         : A
+        p36 = re.compile(r'^Key version\s*: (?P<key_version>\S+)')
+
+        # Configuration last modified by enable_15 at 20:39:39.869 UTC Mon Jun 7 2021
+        p37 = re.compile(r'^Configuration last.+by (?P<last_modified_by>\S+) at '
+                         r'(?P<last_modified_date>[\d:\.]+.+)')
+
+        # Configuration has not been modified since last system restart.
+        p37_1 = re.compile(r'^(?P<last_modified_date>Configuration has not been modified since last system restart.+)')
+
+        # Encryption hardware device : Cisco ASA-55x0 on-board accelerator (revision 0x0)
+        p38 = re.compile(r'^\s*Encryption.+:\s+(?P<encryption_device>.+)')
+
+        # Boot microcode   : CN1000-MC-BOOT-2.00 
+        p39 = re.compile(r'^\s*Boot [Mm]icrocode\s+:\s+(?P<boot_microcode>.+)')
+
+        # SSL/IKE microcode: CNLite-MC-SSLm-PLUS-2.03
+        p40 = re.compile(r'^\s*SSL.+\s*:\s*(?P<ssl_ike_microcode>.+)')
+
+        # IPsec microcode  : CNlite-MC-IPSECm-MAIN-2.06
+        p41 = re.compile(r'^\s*[Ii][Pp][Ss][Ee][Cc].+\s*:\s+(?P<ipsec_microcode>.+)')
+
+        # GTP/GPRS                          : Enabled        perpetual
+        p42 = re.compile(r'^\s*GTP.+: (?P<gtp_gprs>\S+)\s+(?P<time_remaining>\S+)')
+
+        # Maximum Physical Interfaces       : Unlimited      perpetual
+        p43 = re.compile(r'^[Mm]aximum [Pp]hysical [Ii]nterfaces\s*: (?P<max_physical_interfaces>\S+)\s+(?P<time_remaining>\S+)')
+
+        # Shared AnyConnect Premium Peers : 12000          perpetual
+        p44 = re.compile(r'^\s*Shared [Aa]ny[Cc]onnect.+: (?P<shared_anyconnect_premium_peers>\d+)\s+(?P<time_remaining>\S+)')
+
+        # UC Phone Proxy Sessions           : 12             62 days
+        p45 = re.compile(r'^UC Phone.+: (?P<uc_phone_proxy_sessions>\d+)\s+(?P<time_remaining>.+)')
+
+        # Total UC Proxy Sessions           : 12             62 days
+        p46 = re.compile(r'^Total UC Proxy.+: (?P<total_uc_proxy_sessions>\d+)\s+(?P<time_remaining>.+)')
+
+        # Intercompany Media Engine         : Disabled       perpetual
+        p47 = re.compile(r'^Intercompany.+: (?P<intercompany_media_engine>\S+)\s+(?P<time_remaining>.+)')
+
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            # Cisco Adaptive Security Appliance Software Version 9.8(4)10
+            m = p0.match(line)
+            if m:
+                if 'version' not in version_dict:
+                    version_dict['version'] = {}
+                asa_version = m.groupdict()['asa_version']
+                # 9.8(4)10
+                version_dict['version']['asa_version'] = asa_version
```

Comment:
No need to change your code, but I think it's worth mentioning: If you use the `setdefault` method, you can make your code cleaner.
For example:
`
version = version_dict.setdefault('version',  {})
`
Once you have `version`, you can do the same for all the other keys, for example:
`
interfaces_dict = version.setdefault('interfaces', {})
`